### ISSUE TITLE
V2 : createFiberFromTypeAndProps error ( Edit Ranking )

### ISSUE THREAD

**Issue by alsherif-khalaf** [medusajs/medusa#7993]
### Describe the bug

An error occurred while rendering the Medusa Admin page.  The specific error message is "An error occurred while rendering this page" followed by "createFiberFromTypeAndProps@http://localhost:9000/static/js/main.chunk.js".

### System information

Medusa version (including plugins):
Node.js version:
Database:postgresql 16
Operating system: ubuntu 24.04
Browser (if relevant): firefox

### Steps to reproduce the behavior

just installed v2 and go to categorise and click on edit ranking

### Expected behavior

it shoud open a new editing window

### Screenshots

![Screenshot from 2024-07-06 13-15-32](https://github.com/medusajs/medusa/assets/73169053/a5bb7c3b-e1d4-453d-859b-854d6a39d083)

**Comment by alsherif-khalaf** [comment]
using new upgrade :

![image](https://github.com/medusajs/medusa/assets/73169053/3d7a7a60-30ac-4d7c-86fa-f76a6cb6acbb)

![image](https://github.com/medusajs/medusa/assets/73169053/aad8c2c2-fc4d-4009-964f-01a091174e6c)

**Comment by 1ezzy** [comment]
experiencing a similar issue when trying to create a category -- just installed v2, spun up the application, and tried creating a category with no luck. Firefox on macOS using v2 preview, screenshots attached.

<img width="1512" alt="Screenshot 2024-07-18 at 3 20 48 PM" src="https://github.com/user-attachments/assets/98689dbd-7c6c-4c0c-af57-6322334214b8">

**Comment by YoungCR** [comment]
I have the same "createFiberFromTypeAndProps@http: //localhost" from above. Fresh install on Mac OS 14.5

**Comment by codingpandaren** [comment]
Having the same issue when trying to create a category. Using preview version.

```Warning: React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of "CategoryTree".
CategoryTree@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-N6XVZEBD.js?v=905b147e:2021:20
CreateCategoryNesting@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/category-create-M6YVAN7L-K26GDAZN.js?v=905b147e:165:29
div
Primitive</Node2<@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-X6RHCQSI.js?v=905b147e:29021:44
Presence@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-X6RHCQSI.js?v=905b147e:29554:33
TabsContent<@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/ch…```


```Uncaught Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of "CategoryTree".
```


```The above error occurred in the <div> component:

div
CategoryTree@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-N6XVZEBD.js?v=905b147e:2021:20
CreateCategoryNesting@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/category-create-M6YVAN7L-K26GDAZN.js?v=905b147e:165:29
div
Primitive</Node2<@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-X6RHCQSI.js?v=905b147e:29021:44
Presence@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-X6RHCQSI.js?v=905b147e:29554:33
TabsContent<@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/admin-sdk/node_modules/.vite/deps/chunk-X6RHCQSI.js?v=905b147e:59361:75
ProgressTabsContent<@http://localhost:9000/app/@fs/home/matas/codinity-back/node_modules/@medusajs/ad…
```

```
React Router caught the following error during render Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.

Check the render method of `CategoryTree`.
```

**Comment by alsherif-khalaf** [comment]
I tried the latest preview and the problem still exist

**Comment by kasperkristensen** [comment]
Thanks for the report, the issue has been fixed in #8599, and is available in the latest `preview` release.
